massa-async-pool/src/message.rs

```diff
@@ -0,0 +1,219 @@
+//! Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+//! This file defines the structure representing an entry in the FinalLedger
+
+use massa_models::constants::ADDRESS_SIZE_BYTES;
+use massa_models::{
+    array_from_slice, Address, Amount, DeserializeVarInt, ModelsError, SerializeVarInt, Slot,
+};
+use massa_models::{DeserializeCompact, SerializeCompact};
+use serde::{Deserialize, Serialize};
+
+/// Unique identifier of a message.
+/// Also has the property of ordering by priority following the triplet:
+/// (max_gas*gas_price, rev(emission_slot), rev(emission_index))
+pub type AsyncMessageId = (Amount, std::cmp::Reverse<Slot>, std::cmp::Reverse<u64>);
+
+/// Structure defining an asynchronous smart contract message
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AsyncMessage {
+    /// Slot at which the message was emitted
+    pub emission_slot: Slot,
+
+    /// Index of the emitted message within the emission_slot.
+    /// This is used for disambiguating the emission of multiple messages at the same slot.
+    pub emission_index: u64,
```

Comment:
How is this set ?

Reply:
In send_message interface implementation: https://github.com/massalabs/massa/blob/async-pool-integration/massa-execution-worker/src/interface_impl.rs#L469